### PR TITLE
Update OWNERS for tsconfig and compiler

### PR DESCRIPTION
**Pull request by samouri**
**summary**
Marks `wg-performance` as owners of `src/compiler` and `**/tsconfig.json` as we are the authors/maintainers of both.

### CODE REVIEW

File: OWNERS [hunk 1]
```diff
@@ -48,7 +48,7 @@
     },
     {
       // Locked down to minimize new import aliases being added
-      pattern: 'tsconfig{,.base}.json',
+      pattern: '**/tsconfig{,.base}.json',
```
Comment (rcebulko) [inline OWNERS:51]: This means all tsconfig files in all directories require wg-performance approval. May I suggest splitting this into two rules:
1. root `tsconfig{,.base}.json` with `required: true`
2. recursive `**/tsconfig.json` without (also doesn't need the base bit)

Comment (samouri) [reply OWNERS:51]: I'm actually very unfamiliar with how our OWNERS files work. My primary goal is to ensure me & you are able to modify any tsconfig file without needing approval from other working groups. My impression is that we get to a bit lazier than most about it due to working closely with Justin

Comment (samouri) [reply OWNERS:51]: Read up on the docs, LGTM. Making the update

Comment (rcebulko) [reply OWNERS]: Have you seen https://ampproject-owners-bot.appspot.com/example?

**Review by rcebulko — APPROVED** [review]

File: OWNERS (hunk 1 above)
Comment (samouri) [reply OWNERS]: That was the doc I read. Although not the nice HTML version of it, the one in ampthml

**Comment by samouri** [comment]
@jridgewell for OWNERS approval of OWNERS change

**Review by jridgewell — APPROVED** [review]

**Pull request merged**